### PR TITLE
Fix bug allowing metadata edits to e.g. creationTimestamp for built-in types as CRDs

### LINKED ISSUE #1647: Deployment’s creationTimestamp should not be mutable
**Describe the bug**
kuard.yaml metadata is creationTimestamp: null , and kubectl get deployment -o yaml shows metadata creationTimestamp: null , this indicates creationTimestamp is not immutable and if you kubectl edit deployment to any invalid value for example creationTimestamp: "2000-07-29T08:50:59Z" and save, this can be successful!! And the AGE shows 22y

**To Reproduce**
1. Run command 'kubectl create deployment kuard1 --image gcr.io/kuar-demo/kuard-amd64:blue --dry-run=client -o yaml > kuard1.yaml'
2. Now edit kuard.yaml file and set Timestamp to "2000-07-29T08:50:59Z" and save, this can be successful!! And the AGE shows 22y

**Expected behavior**
creationTimestamp field should not be mutable.

**Additional context**

### PR DESCRIPTION
**Pull request by ncdc**
## Summary
Update Kubernetes to pull in https://github.com/kcp-dev/kubernetes/pull/92 to fix an issue where strategic merge patching of e.g. deployments resulted in modifying the original object before field resets happened, meaning fields such as creationTimestamp were mutable when they have to be immutable.

Add an e2e to verify this is behaving as expected.

## Related issue(s)

Fixes #1647

### CODE REVIEW

File: test/e2e/conformance/metadata_test.go [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+	require.NoError(t, err)
+	t.Logf("Verifying creation timestamp was not modified")
+	require.Equal(t, originalCreationTimestamp, patched.GetCreationTimestamp())
+
+}
+
+func jsonOrDie(obj interface{}) []byte {
+	ret, err := json.Marshal(obj)
+	if err != nil {
+		panic(err)
```
Comment (stevekuznetsov) [inline test/e2e/conformance/metadata_test.go:105]: You have `*testing.T`, might as well just call `t.Fatal()` here for DX

File: test/e2e/conformance/metadata_test.go [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+
+	updated := d.DeepCopy()
+	updated.CreationTimestamp.Time = originalCreationTimestamp.Add(-24 * time.Hour * 365)
+
+	patch, err := strategicpatch.CreateTwoWayMergePatch(jsonOrDie(d), jsonOrDie(updated), &appsv1.Deployment{})
+	require.NoError(t, err, "error creating patch")
+
+	t.Logf("Patching deployment - trying to change creation timestamp")
+	patched, err := kubeClusterClient.AppsV1().Deployments("default").Patch(logicalcluster.WithCluster(ctx, workspaceName), d.Name, types.StrategicMergePatchType, patch, metav1.PatchOptions{})
+	require.NoError(t, err)
```
Comment (stevekuznetsov) [inline test/e2e/conformance/metadata_test.go:96]: Does this match k8s behavior?

Comment (ncdc) [reply test/e2e/conformance/metadata_test.go:96]: Against a kind cluster:
```
k -n kcp-1h4f4qj2tgpb patch deploy kuard --type=strategic -p '{"metadata":{"creationTimestamp":"2010-07-29T08:50:59Z"}}' -v 6
I0819 19:36:10.270634   12946 loader.go:372] Config loaded from file:  /Users/angoldst/.kube/config
I0819 19:36:10.297599   12946 round_trippers.go:553] GET https://127.0.0.1:57334/apis/apps/v1/namespaces/kcp-1h4f4qj2tgpb/deployments/kuard 200 OK in 20 milliseconds
I0819 19:36:10.311469   12946 round_trippers.go:553] PATCH https://127.0.0.1:57334/apis/apps/v1/namespaces/kcp-1h4f4qj2tgpb/deployments/kuard?fieldManager=kubectl-patch 200 OK in 11 milliseconds
deployment.apps/kuard patched (no change)
```

**Comment by stevekuznetsov** [comment]
Will LGTM after deps revert

**Review by stevekuznetsov — APPROVED** [review]

**Pull request merged**